### DISCUSSION TITLE
Codegen

### DISCUSSION THREAD

**Discussion by jdx** [jdx/mise#1363] [Ideas]
Here's a user story I have no idea how to fix but curious if anyone has an idea—or simply wants to commiserate.

In general, I like code generation. One example is the [cli reference markdown doc](https://github.com/jdx/mise/blob/main/docs/cli-reference.md). I have never found a great way to build it though. I've used 3 methods:

* "Remembering" to update it by executing a task manually when creating PRs that modify it—I forget this often. I can enforce that the file didn't change in CI but that's a decent amount of infra to setup and annoying when you forget to update it.
* pre-commit hook: this gets slow, quickly. I really don't like using pre-commit hooks at all frankly.
* CI job that writes to the PR. This is my current favorite, but it's kind of a lot to setup. In GHA in particular it creates a new problem that once a job is written to in an action it doesn't execute the actions on the changes so I have to pull down the code and rebase it or something to trigger them.

In addition to that, there is also the codegen process itself: can we make it easy to create files from templates? Is that something that can/should be done by mise or a separate tool?

Has anyone found a way to do codegen that you like in your project? I'm curious what, if anything, mise could do to make this simpler.

Broadly here are some ideas:
* somehow make it easy to synchronize tasks and ci jobs (maybe `mise ci generate --github-actions` to generate ci jobs)?
* pre-commit hooks—as I said I don't love these. I think they might make sense in some narrow and _fast_ scenarios though.

**Comment by BasixKOR** [comment]
Sems like `[[watch_files]]` is one of solutions for this and I like it, but to add my two cents:

- It would be nice if mise comes with built-in templating to generate files, like how it does with `vars` and `env`. My use case is that I'd like to generate a `.xcode.env.local` to point the mise-installed node binary for [React Native environment](https://reactnative.dev/docs/set-up-your-environment#optional-configuring-your-environment) and for now I'd have to write a separate script for that.
- As a continuation of templating idea, it would be nice if we can embed jq filters for template? Parsing `package.json` would be common use case, I think.
